src/Instance/ItemsList/utils.js

```diff
@@ -1,21 +1,109 @@
+    return -1;
+  } else if (bMatches) { // only b is numeric; a > b
+    return 1;
+  }
+
+  // both are strings, or numbers compared equally
+  return a.localeCompare(b);
+};
+
+/**
+ * enumerationComparator
+ * Parse an object's enumeration value numerically by extracting the
+ * first three sets of numbers in the string and comparing them. Further
+ * values are ignored. If are present in only one string, that string sorts
+ * first. If no numbers are present, compare alphabetically.
+ *
+ * @param object a: an object containing the property "enumeration".
+ * @param object b: an object containing the property "enumeration".
+ * @return int < 0 if a is before b, 0 if they are equal, > 0 if a is after b
+ */
+const enumerationComparator = (a, b) => {
+  return alphaNumericComparator((a.enumeration ?? ''), (b.enumeration ?? ''));
+};
+
+/**
+ * loanTypeComparator
+ * Order by temporary loan-type name if available, or permanent loan-type name.
+ *
+ * @param object a: an object containing the property permanentLoanType and/or temporaryLoanType
+ * @param object b: an object containing the property permanentLoanType and/or temporaryLoanType
+ * @return int < 0 if a is before b, 0 if they are equal, > 0 if a is after b
+ */
+const loanTypeComparator = (a, b) => {
+  const aStr = a.temporaryLoanType?.name?.toLowerCase() || a.permanentLoanType?.name?.toLowerCase();
+  const bStr = b.temporaryLoanType?.name?.toLowerCase() || b.permanentLoanType?.name?.toLowerCase();
+  return aStr.localeCompare(bStr);
+};
 
 const sorters = {
-  'barcode': ({ barcode }) => barcode,
-  'status': ({ status }) => status.name.toLowerCase(),
-  'copyNumber': ({ copyNumber }) => copyNumber?.toLowerCase(),
-  'materialType': ({ materialType }) => materialType.name.toLowerCase(),
-  'loanType': (item) => item.temporaryLoanType?.name?.toLowerCase() || item.permanentLoanType?.name?.toLowerCase(),
-  'effectiveLocation': ({ effectiveLocation }) => effectiveLocation.name.toLowerCase(),
-  'enumeration': ({ enumeration }) => enumeration,
-  'chronology': ({ chronology }) => chronology,
-  'volume': ({ volume }) => volume,
-  'yearCaption': ({ yearCaption }) => yearCaption,
+  'barcode': (a, b) => a.barcode.localeCompare(b.barcode),
+  'status': (a, b) => a.status.name.toLowerCase().localeCompare(b.status.name.toLowerCase()),
+  'copyNumber': (a, b) => a.copyNumber?.toLowerCase().localeCompare(b.copyNumber.name.toLowerCase()),
+  'materialType': (a, b) => a.materialType.name.toLowerCase().localeCompare(b.materialType.name.toLowerCase()),
+  'loanType': loanTypeComparator,
+  'effectiveLocation': (a, b) => a.effectiveLocation.name.toLowerCase().localeCompare(b.effectiveLocation.name.toLowerCase()),
+  'enumeration': enumerationComparator,
+  'chronology': (a, b) => a.chronology.localeCompare(b.chronology),
+  'volume': (a, b) => a.volume.localeCompare(b.volume),
+  'yearCaption': (a, b) => a.yearCaption.localeCompare(b.yearCaption),
```

Comment:
I worry a bit that some of these fields are not required and may not be always present. Should we add `?.` to some of them like you did with `copyNumber`?

Reply:
Thanks for pointing this out. I added `?.` where necessary and fixed the `yearCaption` comparator, which was outright broken because `yearCaption` is an array value, not a string value, and therefore can't directly use `localeCompare` without being converted to string first 🤦 .